libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -0,0 +1,659 @@
+        // @Field: PosVarE: position variance east
+        // @Field: PosVarD: position variance down
+        // @Field: VelVarN: velocity variance north
+        // @Field: VelVarE: velocity variance east
+        // @Field: VelVarD: velocity variance down
+
+        AP::logger().WriteStreaming("ILB2", "TimeUS,PosVarN,PosVarE,PosVarD,VelVarN,VelVarE,VelVarD",
+                                    "smmmnnn",
+                                    "F000000",
+                                    "Qffffff",
+                                    now_us,
+                                    state2.kf_pos_covariance.x, state2.kf_pos_covariance.x, state2.kf_pos_covariance.z,
+                                    state2.kf_vel_covariance.x, state2.kf_vel_covariance.x, state2.kf_vel_covariance.z);
+
+        // @LoggerMessage: ILB3
+        // @Description: InertialLabs AHRS data3
+        // @Field: TimeUS: Time since system startup
+        // @Field: Stat1: unit status1
+        // @Field: Stat2: unit status2
+        // @Field: FType: fix type
+        // @Field: SpStat: spoofing status
+        // @Field: GI1: GNSS Info1
+        // @Field: GI2: GNSS Info2
+        // @Field: GJS: GNSS jamming status
+        // @Field: TAS: true airspeed
+        // @Field: WVN: Wind velocity north
+        // @Field: WVE: Wind velocity east
+        // @Field: WVD: Wind velocity down
+
+        AP::logger().WriteStreaming("ILB3", "TimeUS,Stat1,Stat2,FType,SpStat,GI1,GI2,GJS,TAS,WVN,WVE,WVD",
+                                    "s-----------",
+                                    "F-----------",
+                                    "QHHBBBBBffff",
+                                    now_us,
+                                    state2.unit_status, state2.unit_status2,
+                                    state2.gnss_extended_info.fix_type, state2.gnss_extended_info.spoofing_status,
+                                    state2.gnss_info_short.info1, state2.gnss_info_short.info2,
+                                    state2.gnss_jam_status,
+                                    state2.true_airspeed,
+                                    state2.wind_speed.x, state2.wind_speed.y, state2.wind_speed.z);
+    }
+        
+    return true;
+}
+
+void AP_ExternalAHRS_InertialLabs::update_thread()
+{
+    // Open port in the thread
+    uart->begin(baudrate, 1024, 512);
+
+    /*
+      we assume the user has already configured the device
+     */
+
+    setup_complete = true;
+    while (true) {
+        if (!check_uart()) {
+            hal.scheduler->delay_microseconds(250);
+        }
+    }
+}
+
+// get serial port number for the uart
+int8_t AP_ExternalAHRS_InertialLabs::get_port(void) const
+{
+    if (!uart) {
+        return -1;
+    }
+    return port_num;
+};
+
+// accessors for AP_AHRS
+bool AP_ExternalAHRS_InertialLabs::healthy(void) const
+{
+    WITH_SEMAPHORE(state.sem);
+    return AP_HAL::millis() - last_att_ms < 100;
```

Comment:
Do you intend to exclude other state flags like `last_vel_ms`?

Does `state2.unit_status2` encode healthy/iunhealthy?

Reply:
yes, fixed to use those now I have the docs for them